dal/redis.go

```diff
@@ -0,0 +1,57 @@
+package dal
+
+import (
+	"github.com/go-redis/redis"
+)
+
+const ADDR = "localhost:6379"
+
+type Redis struct {
+	Client *redis.Client
+}
+
+func (r *Redis) InitClient() {
+	client := redis.NewClient(&redis.Options{
+		Addr: ADDR,
+	})
+	r.Client = client
+	return
+}
+func (r *Redis) AddActivity(username string, req string) {
+	if !FieldExist(username){
```

Comment:
there is no need to do this validation here - you don't need to call this function if `username` doesn't exist.
(if you want to double-check - just check if `username == ""`